src/ert/async_utils.py

```diff
@@ -26,22 +26,26 @@ async def background_tasks() -> AsyncGenerator[Any, Any]:
 
     """
 
-    tasks: MutableSequence[asyncio.Task[Any]] = []
+    tasks: List[asyncio.Task[Any]] = []
+    gather_task = asyncio.gather(*tasks, return_exceptions=True)
 
     def add(coro: Coroutine[Any, Any, Any]) -> None:
-        tasks.append(asyncio.create_task(coro))
+        tasks.append(_create_task(get_event_loop(), coro))
 
     try:
         yield add
     finally:
         for t in tasks:
             t.cancel()
-        for exc in await asyncio.gather(*tasks, return_exceptions=True):
-            if isinstance(exc, asyncio.CancelledError):
-                continue
-            if isinstance(exc, BaseException):
-                logger.error(str(exc), exc_info=exc)
-        tasks.clear()
+        try:
+            for exc in await gather_task:
+                if not isinstance(exc, BaseException):
+                    continue
+                raise exc
+        except asyncio.CancelledError:
```

Comment:
Thinking out loud: Do we risk stumbling on a CancelledError before we have gone through all tasks, and then miss an exception?

Reply:
I assume that if we stumble upon an exception after cancellederror, it would be due to something not working because something else was cancelled.

Reply:
No, you're right! What if a task fails, cancels the other tasks, but the cancellation of the other tasks finish before the exception is propagated? I guess this is how we have to do it, otherwise something could be hanging.